react-components/src/hooks/useClickedNode.test.tsx

```diff
@@ -0,0 +1,322 @@
+      expect(result.current).toEqual({
+        mouseButton: MOUSE.LEFT,
+        position: new Vector2(100, 200),
+        fdmResult: undefined,
+        assetMappingResult: undefined,
+        pointCloudAnnotationMappingResult: mockPointCloudData,
+        pointCloudFdmVolumeMappingResult: null,
+        intersection: createMockCadIntersection()
+      });
+    });
+  });
+
+  test('returns point cloud FDM volume mapping result when intersection and data are available', async () => {
+    const mockVolumeData = [
+      { assetInstance: { space: 'test', externalId: 'volume-1' }, views: [] }
+    ];
+
+    viewerGetAnyIntersectionFromPixelMock.mockResolvedValue(createMockCadIntersection());
+
+    mockUsePointCloudFdmVolumeMappingForIntersection.mockReturnValue(
+      createMockQueryResult(mockVolumeData, false)
+    );
+
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(clickCallback, MOUSE.LEFT);
+
+    await waitFor(async () => {
+      expect(result.current).not.toBeUndefined();
+      expect(result.current).toEqual({
+        mouseButton: MOUSE.LEFT,
+        position: new Vector2(100, 200),
+        fdmResult: undefined,
+        assetMappingResult: undefined,
+        pointCloudAnnotationMappingResult: null,
+        pointCloudFdmVolumeMappingResult: mockVolumeData,
+        intersection: createMockCadIntersection()
+      });
+    });
+  });
+
+  test('sets up event listeners for 360 annotation intersection handling', async () => {
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    expect(viewer.on).toHaveBeenCalledWith('click', expect.any(Function));
+    expect(result.current).toBeUndefined();
+  });
+
+  test('returns undefined when no intersection data is available', () => {
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    expect(result.current).toBeUndefined();
+  });
+
+  test('respects leftClick option when provided', async () => {
+    viewerGetAnyIntersectionFromPixelMock.mockResolvedValue(createMockCadIntersection());
+
+    const { result } = renderHook(() => useClickedNodeData({ leftClick: false }), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(clickCallback, MOUSE.LEFT);
+
+    expect(result.current).toBeUndefined();
+  });
+
+  test('respects rightClick option when provided', async () => {
+    viewerGetAnyIntersectionFromPixelMock.mockResolvedValue(createMockCadIntersection());
+
+    const { result } = renderHook(() => useClickedNodeData({ rightClick: true }), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(clickCallback, MOUSE.RIGHT);
+
+    await waitFor(async () => {
+      expect(result.current).not.toBeUndefined();
+      expect(result.current).toEqual({
+        mouseButton: MOUSE.RIGHT,
+        position: new Vector2(100, 200),
```

Comment:
I would prefer to have the position in a variable and also pass it to `simulateClick`